--- a/net/minecraft/server/ServerAdvancementManager.java
+++ b/net/minecraft/server/ServerAdvancementManager.java
@@ -26,7 +_,12 @@
     private AdvancementTree tree = new AdvancementTree();
     private final HolderLookup.Provider registries;
 
+    /** @deprecated Forge: use {@linkplain ServerAdvancementManager#ServerAdvancementManager(net.minecraftforge.common.crafting.conditions.ICondition.IContext) constructor with context}. */
     public ServerAdvancementManager(HolderLookup.Provider p_336198_) {
+        this(p_336198_, net.minecraftforge.common.crafting.conditions.ICondition.IContext.EMPTY);
+    }
+
+    public ServerAdvancementManager(HolderLookup.Provider p_336198_, net.minecraftforge.common.crafting.conditions.ICondition.IContext context) {
         super(p_336198_, Advancement.CODEC, Registries.ADVANCEMENT);
         this.registries = p_336198_;
     }
